File: python/program/Fan.py
from Komponen import Komponen


class Fan(Komponen):
    def __init__(self, kecepatan=0, nama="", merk=""):
        Komponen.__init__(self, merk, nama)
        self.kecepatan = kecepatan
        self.status = "off"

    def set_kecepatan(self, kecepatan):
        self.kecepatan = kecepatan

    def turn_on(self):
        self.status = "on"

    def turn_off(self):
        self.status = "off"

    def get_kecepatan(self):
        return self.kecepatan

    def get_status(self):
        return self.status

File: cpp/program/Fan.cpp
#include "Komponen.cpp"
#pragma once
#include <string>

class Fan : public Komponen {
  private:
    int kecepatan;
    string status;

  public:
    Fan() {
        this->status = "off";
    }

    Fan(int kecepatan, string nama, string merk)
        : Komponen(merk, nama) {
        this->kecepatan = kecepatan;
        this->status = "off";
    }

    void setKecepatan(int kecepatan) {
        this->kecepatan = kecepatan;
    }

    void TurnOn() {
        this->status = "on";
    }

    void TurnOff() {
        this->status = "off";
    }

    int getKecepatan() {
        return this->kecepatan;
    }

    string getStatus() {
        return this->status;
    }

    ~Fan() {}
};
